Rework plot computations

n-koenig opened the pull request:
Includes new plot functions `plot_indicators()` and `plot_comp()`. Reworked `get_indicators()` to consolidate retrieval of indicator information.

The new `get_indicators()` method is meant to replace `identify_indicator_timings()`. Per default `get_indicators()` returns the same information as before. But it also offers new formats which are (currently) only used by `plot_indicators()` and `plot_comp()`, but could also be used by other plot functions.

Tests for the new functions are still missing at the moment.

n-koenig committed:
Merge plot_comp_ebba into dev

Squashed commit of the following:

commit 5bad1c0892a3f5a10aa1b4ec868f8de208d2dc96
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 12:52:32 2025 +0200

    Revert roxygen2 version

commit 1061921a5b625281c9a5ad307db608a48b9e3fc3
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 12:41:54 2025 +0200

    Revert plot_comp to only plot non-cumulative fesis breaks

commit 938b2cccae2d096d003d3193f201c945eca0c8a4
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 11:17:32 2025 +0200

    Update documentation and github workflow, remove plot_comp_quick

commit 61134268180399d497041fbff10cf901654bd94d
Author: Nils König <nils.koenig01@gmail.com>
Date:   Mon Jun 23 15:32:16 2025 +0200

    Add roxygen outputs

commit f725aba83bc8ca9553cf5cfb4180dc774c9368d2
Author: Nils König <nils.koenig01@gmail.com>
Date:   Mon Jun 23 15:26:21 2025 +0200

    Add compare functions to namespace

commit 4ec03f5ef5253f05284dcf98a634dfb7ee0171aa
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed Jun 18 18:51:48 2025 +0200

    Implement fully functional expansion of break list to panel coef table to also support iis and tis

commit 51c2867402a02bb4e1203ac8fe6f16e347e3276d
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed Jun 18 16:21:46 2025 +0200

    Use gets::coef to get the coefficients faster and calculate cumulative coefficients

commit 3a3e7fd7ffd76a8242a4a4dec6d500c69445ca0b
Author: ebbam <99182071+ebbam@users.noreply.github.com>
Date:   Mon Sep 16 16:22:35 2024 +0100

    plot_comp funs and example res file

n-koenig committed:
Merge plot_indicators into dev

Squashed commit of the following:

commit 8f267901973cbc79d90735aaf72605bed280bfc8
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 13:13:52 2025 +0200

    Update with roxygen

commit 82c5e6e5bc7860148d7881055c7c0c3c418504d2
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 17 11:52:10 2025 +0200

    Correct fixed effects plotting, some better descriptions

commit 5f3b54f450bf82393c3dd70a658a2aca0cbb618d
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed Jun 11 11:43:55 2025 +0200

    Re-implement option to include fixed effects and some more cleaning

commit 42bf1b0610cd48234864d6965c6070effb650b05
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 10 15:00:49 2025 +0200

    Cleanup and rework plot_indicators

commit a1d2101e2145c50b34f39de1db0964b4b9353526
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed May 28 13:48:22 2025 +0200

    First Version of Indicator Plotting Function

n-koenig committed:
Merge plot_comp_ebba  into dev

Squashed commit of the following:

commit f31ba12d64ec467852036f9cf4d86ef773844578
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 13:37:43 2025 +0200

    Update roxygen imports

commit 5bad1c0892a3f5a10aa1b4ec868f8de208d2dc96
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 12:52:32 2025 +0200

    Revert roxygen2 version

commit 1061921a5b625281c9a5ad307db608a48b9e3fc3
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 12:41:54 2025 +0200

    Revert plot_comp to only plot non-cumulative fesis breaks

commit 938b2cccae2d096d003d3193f201c945eca0c8a4
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 11:17:32 2025 +0200

    Update documentation and github workflow, remove plot_comp_quick

commit 61134268180399d497041fbff10cf901654bd94d
Author: Nils König <nils.koenig01@gmail.com>
Date:   Mon Jun 23 15:32:16 2025 +0200

    Add roxygen outputs

commit f725aba83bc8ca9553cf5cfb4180dc774c9368d2
Author: Nils König <nils.koenig01@gmail.com>
Date:   Mon Jun 23 15:26:21 2025 +0200

    Add compare functions to namespace

commit 4ec03f5ef5253f05284dcf98a634dfb7ee0171aa
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed Jun 18 18:51:48 2025 +0200

    Implement fully functional expansion of break list to panel coef table to also support iis and tis

commit 51c2867402a02bb4e1203ac8fe6f16e347e3276d
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed Jun 18 16:21:46 2025 +0200

    Use gets::coef to get the coefficients faster and calculate cumulative coefficients

commit 3a3e7fd7ffd76a8242a4a4dec6d500c69445ca0b
Author: ebbam <99182071+ebbam@users.noreply.github.com>
Date:   Mon Sep 16 16:22:35 2024 +0100

    plot_comp funs and example res file

n-koenig committed:
Merge plot_indicators into dev

Squashed commit of the following:

commit 1bdd40723842b769ee77ac32eb6daad5e3891d26
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 13:31:37 2025 +0200

    Some more roxygen imports

commit 8f267901973cbc79d90735aaf72605bed280bfc8
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 24 13:13:52 2025 +0200

    Update with roxygen

commit 82c5e6e5bc7860148d7881055c7c0c3c418504d2
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 17 11:52:10 2025 +0200

    Correct fixed effects plotting, some better descriptions

commit 5f3b54f450bf82393c3dd70a658a2aca0cbb618d
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed Jun 11 11:43:55 2025 +0200

    Re-implement option to include fixed effects and some more cleaning

commit 42bf1b0610cd48234864d6965c6070effb650b05
Author: Nils König <nils.koenig01@gmail.com>
Date:   Tue Jun 10 15:00:49 2025 +0200

    Cleanup and rework plot_indicators

commit a1d2101e2145c50b34f39de1db0964b4b9353526
Author: Nils König <nils.koenig01@gmail.com>
Date:   Wed May 28 13:48:22 2025 +0200

    First Version of Indicator Plotting Function

n-koenig committed:
Change plot_comp to display cumulative coefficients

n-koenig committed:
Added axis labels to plot_comp

n-koenig committed:
Add regex_exclude_indicators parameter to plot_comp

n-koenig committed:
Fix naming error

n-koenig committed:
Rework get_indicators to provide table or long format (for plotting) while staying compatible with list format

n-koenig committed:
Add coefficients and combined indicator to get_indicators

n-koenig committed:
Change plot_comp and plot_indicators to use get_indicators output and remove tidyr/dplyr dependencies

n-koenig committed:
Update documentation and cleanup for get_indicators

n-koenig committed:
Enhance documentation and input checks for get_indicators

n-koenig committed:
Add regex_exclude_indicators to get_indicators

n-koenig committed:
Documentation and cleanup for plot_indicators

n-koenig committed:
Cleanup, documentation, and validation for plot_comp, removed plot_unit

n-koenig committed:
Add sign to plot_indicators, some documentation updates

n-koenig committed:
Update Description and Namespace

n-koenig committed:
Fix tests/issues and minor updates to get_indicators

n-koenig committed:
Update references to get_indicators

n-koenig committed:
Add better example data and fix warnings

n-koenig committed:
Update documentation

n-koenig committed:
Add capability to handle "/" date formats and more fixes

n-koenig commented on R/get_indicators.R:
```diff
@@ -1,11 +1,56 @@
+#'   If NULL (default), user-specified indicators are not included in the result.
+#' @param format A character string indicating the format of the output. Must be one of:
+#'   \itemize{
+#'     \item "list" (default): Returns a list of data frames for backward compatibility
```
I changed list and table to have the same columns now, so list is probably still the better default because people are used to it and it's easier to differentiate the break types

n-koenig replied:
I removed the "legacy" mentions so documentation sounds better now

n-koenig commented on R/get_indicators.R:
```diff
@@ -32,24 +77,192 @@
+  all_indicators_long <- merge(all_indicators_long, coefficients, by = c("name"), all.x = TRUE)
+
+  # Initialize output based on format
+  output <- if (format == "list") list() else data.frame()
```
Done!

n-koenig committed:
Requested fixes

n-koenig commented on R/get_indicators.R:
```diff
@@ -1,11 +1,56 @@
+#'   \item time: The panel time period
+#'   \item name: The name of the indicator. For compatibility with `plot.isatpanel`, CFESIS and CSIS indicators are renamed to their variable name in the list format.
+#'   \item type: The type of the indicator (e.g., "IIS", "FESIS", "TIS", "CFESIS", "CSIS", "UIS")
+#'   \item coef: The coefficient of the indicator
```
I added sd to table/list now for sake of completeness, maybe we can discuss next time whether it makes sense in the long format as well

n-koenig committed:
Forgot to run check locally

n-koenig committed:
Fix test again

n-koenig committed:
Added tests for get_indicators and two fixtures for quicker tests

n-koenig committed:
Implement include_blanks correctly in plot_compare_grid

n-koenig committed:
Fix bug when get_indicators returns empty object

n-koenig committed:
Fix sign parameter implementation

n-koenig committed:
Add all tests

n-koenig committed:
Fix note on unknown variable name

moritzpschwarz commented on R/plot_compare_grid.R:
```diff
@@ -0,0 +1,180 @@
+#' data(compare_models_example, package = "getspanel")
+#'
+#' # Basic comparison plot
+#' plot_comp(compare_models_example)
```
Should it be `plot_comp` here?

moritzpschwarz committed:
Update documentation and indicator handling

Improves documentation for data sources and corrects typos in R/data.R and Rd files. Updates get_indicators to infer user-specified indicators when uis_breaks is NULL, fixes handling of UIS results, and ensures proper use of stats namespace. Adjusts plot_indicators to only show legend entries for present indicator types. Adds formula to isatpanel output arguments.
